ddsketch/store/store.go

```diff
@@ -40,14 +48,18 @@ type Store interface {
 	ToProto() *sketchpb.Store
 }
 
-// Returns an instance of DenseStore that contains the data in the provided protobuf representation.
+// FromProto returns an instance of DenseStore that contains the data in the provided protobuf representation.
 func FromProto(pb *sketchpb.Store) *DenseStore {
 	store := NewDenseStore()
+	populateStoreFromProto(pb, store)
+	return store
+}
+
+func populateStoreFromProto(pb *sketchpb.Store, emptyStore Store) {
 	for idx, count := range pb.BinCounts {
-		store.AddWithCount(int(idx), count)
+		emptyStore.AddWithCount(int(idx), count)
 	}
 	for idx, count := range pb.ContiguousBinCounts {
-		store.AddWithCount(idx+int(pb.ContiguousBinIndexOffset), count)
+		emptyStore.AddWithCount(idx+int(pb.ContiguousBinIndexOffset), count)
 	}
-	return store
```

Comment:
This may not be the most efficient way to deserialize for some store types, but I think the authors of those stores would know best.